### PR TITLE
Implementing Bagging Algorithms in Python

### LINKED ISSUE #6189: [Machine Learning] Implementing Bagging Algorithms in Python
#### Proposed title of article
[Machine Learning] Implementing Bagging Algorithms in Python

#### Proposed article introduction
Bagging, also known as bootstrap aggregation, is the ensemble learning method that is commonly used to reduce variance within a noisy dataset. In bagging, a random sample of data in a training set is selected with replacement, this means that the individual data points can be chosen more than once.

After several data samples are generated, these weak models are then trained independently. Depending on the type of task (regression or classification) the average or majority of those predictions yield a more accurate estimate.  Bagging algorithms create several subsets of data from training samples chosen randomly with replacement. Each collection of subset data is used to train their decision trees.

In this tutorial, we will use the `BaggingClassifier` algorithm to train our model. We will build a diabetes prediction model. `BaggingClassifier` algorithm will fine-tune the model until we have a model with the best accuracy score.

#### Key takeaways 
1. Data preprocessing.
2. What is a bagging algorithm?
3. Initializing the `BaggingClassifier` algorithm.
4. Using the `BaggingClassifier` algorithm to build a diabetes prediction model.
5. Comparing the results when using an individual algorithm such as Decision Tree and Random Forest algorithms.

#### Article quality
This tutorial is unique because we will use two approaches in building the model. The first approach is to use an individual algorithm such as the Decision tree algorithm or the Random Forest algorithm in building the model. The second approach is to build the same model using the  `BaggingClassifier` algorithm. We then compare the accuracy scores of the two approaches. This will enable us to see how the Bagging algorithm improves the model's accuracy score. 

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by josephgatura328**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6189

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Review by jerimkaura — COMMENTED** [review]
Hello @josephgatura328 :wave:

Welcome to Section Engineering Education :tada:  As a fellow python developer. I am very excited to have you onboard :rocket: I will be your reviewer and feel free to ask anything related to EngEd. 

Our review process starts from grammar and format checks to code review. Do not worry, I will take you through the process. Kindly be patient as I go over the first round of revision.

Thanks!

**Comment by LinusMuema** [comment]
plagiarism check done ✅

File: content/articles/implementing-bagging-algorithms-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+- [Accuracy score](#accuracy-score)
+- [Building the model using Random Forest Classifier](#building-the-model-using-random-forest-classifier)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Prerequisites
+A reader should know the following:
+
+- [Python programming](/engineering-education/python-projects-for-beginners/)
+- [Machine learning workflows](/engineering-education/house-price-prediction/)
```
Comment (jerimkaura) [inline content/articles/implementing-bagging-algorithms-in-python/index.md:37]: Provide full URLs for these links.

File: content/articles/implementing-bagging-algorithms-in-python/index.md [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+> NOTE: In this tutorial, we will use [Google Colab notebook](https://research.google.com/)
+
+### Bagging vs Boosting
+As mentioned above,  in bagging, multiple homogenous algorithms are trained independently in parallel, while in boosting, multiple homogenous algorithms are trained sequentially. The image below shows the difference between bagging and boosting.
+
+![Bagging Vs Boosting](/engineering-education/implementing-bagging-algorithms-in-python/bagging-vs-boosting.jpg)
+
+*Image Source: [Pluralsight](https://pluralsight2.imgix.net/guides/81232a78-2e99-4ccc-ba8e-8cd873625fdf_2.jpg)*
+
+In this tutorial, we will only be focusing on implementing bagging algorithms. To implement boosting algorithms, read this [article](/engineering-education/boosting-algorithms-python/)
```
Comment (jerimkaura) [inline content/articles/implementing-bagging-algorithms-in-python/index.md:50]: Provide a full URL here.

**Review by jerimkaura — COMMENTED** [review]
Hello @josephgatura328 

Thank you for such a great article. However, kindly provide a full URL for articles referenced from the section's pool of resources. 

Secondly, go over the [sample author page template](https://github.com/section-engineering-education/engineering-education/edit/master/content/authors/jerim-kaura/index.md) and format your according to the standard format.

Work on these so that we process to the next round of revision.

Thanks

**Comment by hectorkambow** [comment]
Good afternoon and thank you for submitting your topic to the EngEd program.
Upon a quick glance we noticed that this articles has some grammatical errors (or is a bit difficult to read) and does not seem ready to be published.

In order to keep our backlog to a minimum, we will be closing this PR, as all edits should be completed on authors forked repo not in the queue.
